### PR TITLE
feat(client): Consolidate HTTP Client

### LINKED ISSUE #2919: Consolidate HTTP client usage
### Summary

As per comment here https://github.com/kumahq/kuma/pull/2892#discussion_r723859457
We could consolidate the verbosity of the raw Go HTTP client.

### PR DESCRIPTION
**Pull request by mmorel-35**
### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues -- closes #2919 
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/util/http/client.go [hunk 1]
```diff
@@ -1,23 +1,73 @@
...
 
-func (f ClientFunc) Do(req *nethttp.Request) (*nethttp.Response, error) {
+func (f ClientFunc) Do(req *http.Request) (*http.Response, error) {
 	return f(req)
 }
 
+type TokenClient struct {
+	client Client
+	url    string
+}
```
Comment (jakubdyszkiewicz) [inline pkg/util/http/client.go:29]: IMHO this is too specific to have this in `pkg/util/http`.
I consider `pkg/util` a package that has a code that the standard library/libaries that we depend lacks in functionality.

How about we move this to `pkg/tokens/client.go` ?

**Review by jakubdyszkiewicz — COMMENTED** [review]
nice refactor!

File: pkg/util/http/client.go (hunk 1 above)
Comment (mmorel-35) [reply pkg/util/http/client.go]: Nice, it done :) !

**Comment by slonka** [comment]
@jakubdyszkiewicz can you re-review?

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**